Packs/Palo_Alto_Networks_Security_Advisories/Integrations/PaloAltoNetworksSecurityAdvisories/PaloAltoNetworksSecurityAdvisories.py

```diff
@@ -0,0 +1,229 @@
+            timeout=self.api_timeout,
+            params=params
+        )
+
+
+def dataclass_to_command_results(result: Any, raw_response: Union[List, Dict]) -> CommandResults:
+    """Takes a list, or a single, dataclass instance and converts it to a CommandResult instance."""
+    if not result:
+        command_result = CommandResults(
+            readable_output="No results.",
+        )
+        return command_result
+
+    outputs: Union[List[Dict], Dict, Any]
+    if type(result) is list:
+        outputs = [vars(x) for x in result]
+        summary_list = [vars(x) for x in result]
+        title = result[0]._title
+        output_prefix = result[0]._output_prefix
+    else:
+        outputs = vars(result)
+        summary_list = [vars(result)]
+        title = result._title
+        output_prefix = result._output_prefix
+
+    extra_args = {}
+    if hasattr(result, "_outputs_key_field"):
+        extra_args["outputs_key_field"] = getattr(result, "_outputs_key_field")
+
+    readable_output = tableToMarkdown(title, summary_list)
+    command_result = CommandResults(
+        outputs_prefix=output_prefix,
+        outputs=outputs,
+        readable_output=readable_output,
+        raw_response=raw_response,
+        **extra_args
+    )
+    return command_result
+
+
+@dataclass
+class Advisory:
+    """
+    :param data_type: The type of advisory this is
+    :param data_format: The format of the advisory, such as MITRE
+    :param cve_id: The ID of the CVE described by this advisory
+    :param cve_date_public: The date this CVE was released
+    :param cve_title: The name of this CVE
+    :param affects_product_name: The name of the product this affects, such as PAN-OS
+    :param description: Human readable description of Advisory
+    :param affected_version_list: List of affected versions strings
+    """
+    data_type: str
+    data_format: str
+    cve_id: str
+    cve_date_public: str
+    cve_title: str
+    affects_vendor_name: str
+    description: str
+    cvss_score: int
+    cvss_severity: str
+    cvss_vector_string: str
+    affected_version_list: list
+
+    _output_prefix = OUTPUT_PREFIX + "Advisory"
+    _title = "Palo Alto Networks Security Advisories"
+
+
+class SeverityEnum(enum.Enum):
+    HIGH = "HIGH"
+    CRITICAL = "CRITICAL"
+    MEDIUM = "MEDIUM"
+    LOW = "LOW"
+    NONE = "NONE"
+
+
+def locals_to_dict(locals_data: dict) -> dict:
+    """Removes all arguments with None values from the dictionary returned by locals()"""
+    result_dict = {}
+    for key, value in locals_data.items():
+        if value:
+            result_dict[key] = value
+
+    return result_dict
+
+
+def flatten_advisory_dict(advisory_dict: dict) -> Advisory:
+    """Given a dictionary advisory, return an `Advisory` object"""
+    affects_dict = {}
+
+    for vendor_data_dict in advisory_dict.get("affects", {}).get("vendor", {}).get("vendor_data", []):
+        affects_dict["affects_vendor_name"] = vendor_data_dict.get("vendor_name")
```

Comment:
This code only saves the last vendor name. It looks like a bug.
If not, please add an inline comment to avoid future confusion.

Reply:
Removed affects_vendor_name - the website only publishes CVEs for Pan, and if the user wants the entire CVE they can use raw response. Best to keep it simple.